src/devices/uFire_ISE/UFireIse.cs

```diff
@@ -0,0 +1,388 @@
+            }
+        }
+
+        /// <summary>
+        /// Low level reads of EEPROM
+        /// </summary>
+        /// <param name="address">Address to read</param>
+        /// <returns>Low level data from EEPROM</returns>
+        public float ReadEEPROM(float address)
+        {
+            if (address < 0xff)
+            {
+                Write_register((byte)Register.ISE_SOLUTION_REGISTER, address);
+                Send_command((byte)Command.ISE_MEMORY_READ);
+                return Read_register((byte)Register.ISE_BUFFER_REGISTER);
+            }
+
+            return 0;
+        }
+
+        /// <summary>
+        /// Low level write of EEPROM
+        /// </summary>
+        /// <param name="address">Address</param>
+        /// <param name="data">Data to write to the address</param>
+        public void WriteEEPROM(float address, float data)
+        {
+            if (address < 0xff)
+            {
+                Write_register((byte)Register.ISE_SOLUTION_REGISTER, address);
+                Write_register((byte)Register.ISE_BUFFER_REGISTER, data);
+                Send_command((byte)Command.ISE_MEMORY_WRITE);
+            }
+        }
+
+        /// <summary>
+        /// Get Firmware version
+        /// </summary>
+        /// <returns>The version of the firmware</returns>
+        public Version GetFirmwareVersion()
+        {
+            return new Version(Read_byte((byte)Register.ISE_FW_VERSION_REGISTER).ToString());
+
+        }
+
+        private void Send_command(byte data)
+        {
+            Span<byte> bytes = stackalloc byte[2];
+            bytes[0] = (byte)Register.ISE_TASK_REGISTER;
+            bytes[1] = data;
+            _device.Write(bytes);
+            DelayHelper.DelayMilliseconds(ISE_COMMUNICATION_DELAY, allowThreadYield: true);
+        }
+
+        private float Read_register(byte register)
+        {
+            byte[] data = { 0, 0, 0, 0 };
+
+            Change_register(register);
+
+            data[0] = _device.ReadByte();
+            data[1] = _device.ReadByte();
+            data[2] = _device.ReadByte();
+            data[3] = _device.ReadByte();
+
+            return Convert.ToSingle(Round_total_digits(BitConverter.ToSingle(data, 0), 7));
+        }
+
+        private void Write_register(byte register, float? data)
+        {
+            byte[] dataArray = { 0, 0, 0, 0 };
+
+            if (data != null)
+            {
+                dataArray = BitConverter.GetBytes(Round_total_digits(data.Value));
+            }
+
+            Span<byte> bytes = stackalloc byte[4];
+            bytes[0] = dataArray[0];
+            bytes[1] = dataArray[1];
+            bytes[2] = dataArray[2];
+            bytes[3] = dataArray[3];
+
+            Change_register(register);
+            _device.Write(bytes);
+            DelayHelper.DelayMilliseconds(ISE_COMMUNICATION_DELAY, allowThreadYield: true);
+        }
+
+        private void Change_register(byte register)
```

Comment:
Please refer to the naming convention, it should be ```ChangeRegister```
Please adjust as well all other functions and names like this